x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformTaskTests.java

```diff
@@ -241,4 +254,152 @@ public void testStopOnFailedTaskWithoutIndexer() {
+                                .addTask("transform-3", TransformTaskParams.NAME, null, null)
+                                .addTask("other-3", "other", null, null)
+                                .build()
+                        )
+                )
+                .build();
+            assertThat(
+                TransformTask.findAllTransformTasks(clusterState).stream().map(PersistentTask::getId).collect(toList()),
+                containsInAnyOrder("transform-1", "transform-2", "transform-3")
+            );
+        }
+    }
+
+    public void testFindTransformTasks() {
+        ClusterState clusterState = ClusterState.builder(new ClusterName("some-cluster"))
+            .metadata(
+                Metadata.builder()
+                    .putCustom(
+                        PersistentTasksCustomMetadata.TYPE,
+                        PersistentTasksCustomMetadata.builder()
+                            .addTask("transform-1", TransformTaskParams.NAME, newTransformTaskParams("transform-1"), null)
+                            .addTask("other-1", "other", null, null)
+                            .addTask("transform-2", TransformTaskParams.NAME, newTransformTaskParams("transform-2"), null)
+                            .addTask("other-2", "other", null, null)
+                            .addTask("transform-3", TransformTaskParams.NAME, newTransformTaskParams("transform-3"), null)
+                            .addTask("other-3", "other", null, null)
+                            .build()
+                    )
+            )
+            .build();
+        assertThat(
+            TransformTask.findTransformTasks("trans*-*", clusterState).stream().map(PersistentTask::getId).collect(toList()),
+            containsInAnyOrder("transform-1", "transform-2", "transform-3")
+        );
+        assertThat(
+            TransformTask.findTransformTasks("*-2", clusterState).stream().map(PersistentTask::getId).collect(toList()),
+            contains("transform-2")
+        );
+        assertThat(TransformTask.findTransformTasks("*-4", clusterState), is(empty()));
+        assertThat(
+            TransformTask.findTransformTasks(Set.of("transform-1", "transform-2", "transform-3", "transform-4"), clusterState)
+                .stream()
+                .map(PersistentTask::getId)
+                .collect(toList()),
+            containsInAnyOrder("transform-1", "transform-2", "transform-3")
+        );
+        assertThat(
+            TransformTask.findTransformTasks(Set.of("transform-1", "transform-2"), clusterState)
+                .stream()
+                .map(PersistentTask::getId)
+                .collect(toList()),
+            containsInAnyOrder("transform-1", "transform-2")
+        );
+        assertThat(TransformTask.findTransformTasks(Set.of("transform-4", "transform-5"), clusterState), is(empty()));
+    }
+
+    private static TransformTaskParams newTransformTaskParams(String transformId) {
```

Comment:
nit: it feels like a missing space, so it is a bit confusing, maybe `createTransformTaskParams`?